### DISCUSSION TITLE
HELP: Use with AdGuard Home on same machine

### DISCUSSION THREAD

**Discussion by Free-Radical** [RaspAP/raspap-webgui#1629] [General]
ADGuard which is a really nice adblocker, and fundamentally a DNS server, when running on the same host creates issues w/ DNSmasq, i.e rasp stops working.

I tried "mask-ing" the DNSmasq service but worked momentarily  but when RaspAd.service ran again it noticed that DNSmasq was "masked" and everything stopped working.

Can somebody pls tell me how to get these 2 work together w/out changing Adguard's use of port 53, pls? (my router wont let me specifiy another port for DNS so need to keep  Adguard running on 53"

**Comment by billz** [comment]
The official method described in the [FAQ](https://docs.raspap.com/faq/#can-i-integrate-raspap-with-adguard-home) is still the simplest and most reliable way to integrate these services.

Changing the dnsmasq port will render RaspAP non-functional; same is also true if you mask/disable this service. If you can't upgrade your router's firmware or use another device  for some reason, you could try redirecting these requests to a different port.

```
iptables -t nat -A OUTPUT -p udp --dport 53 -j DNAT --to some.ip:5353
iptables -t nat -A OUTPUT -p tcp --dport 53 -j DNAT --to some.ip:5353
```

Bear in mind this is untested and therefore totally unsupported.

**Comment by Free-Radical** [comment]
OK Thank you! 

PS Great software, truly appreciate it.
